src/components/Indicator.tsx

```diff
@@ -0,0 +1,113 @@
+import React from 'react';
+import {StyleSheet, View} from 'react-native';
+import {OnyxCollection, withOnyx} from 'react-native-onyx';
+import {OnyxEntry} from 'react-native-onyx/lib/types';
+import * as PolicyUtils from '@libs/PolicyUtils';
+import * as UserUtils from '@libs/UserUtils';
+import useTheme from '@styles/themes/useTheme';
+import useThemeStyles from '@styles/useThemeStyles';
+import * as PaymentMethods from '@userActions/PaymentMethods';
+import ONYXKEYS from '@src/ONYXKEYS';
+import type {BankAccountList, FundList, LoginList, Policy, PolicyMembers, ReimbursementAccount, UserWallet, WalletTerms} from '@src/types/onyx';
+
+type CheckingMethod = () => boolean;
+
+type IndicatorOnyxProps = {
+    /** The employee list of all policies (coming from Onyx) */
+    allPolicyMembers: OnyxCollection<PolicyMembers>;
+
+    /** All the user's policies (from Onyx via withFullPolicy) */
+    policies: OnyxCollection<Policy>;
+
+    /** List of bank accounts */
+    bankAccountList: OnyxEntry<BankAccountList>;
+
+    /** List of user cards */
+    fundList: OnyxEntry<FundList>;
+
+    /** The user's wallet (coming from Onyx) */
+    userWallet: OnyxEntry<UserWallet>;
+
+    /** Bank account attached to free plan */
+    reimbursementAccount: OnyxEntry<ReimbursementAccount>;
+
+    /** Information about the user accepting the terms for payments */
+    walletTerms: OnyxEntry<WalletTerms>;
+
+    /** Login list for the user that is signed in */
+    loginList: OnyxEntry<LoginList>;
+};
+
+type IndicatorProps = IndicatorOnyxProps;
+
+function Indicator({
+    reimbursementAccount = {},
+    allPolicyMembers = {},
+    policies = {},
+    bankAccountList = {},
+    fundList = {},
+    userWallet = null,
+    walletTerms = {},
+    loginList = {},
```

Comment:
I'm not sure if setting empty objects as defaults is a good pattern.  We can use optional chaining.
I think an empty object is not a proper shape of particular prop.

Reply:
These defaults were there before, though I agree it should be okay with the logic since optional chaining was added where necessary.
@barttom Do you think it's better to remove defaults totally then? ^^ @fabioh8010

Reply:
If You don't need to follow previous implementation on 100% and have some space for improvements - please remove them :)

Reply:
I'm okay with removing if the code can handle it 🙂